### ISSUE TITLE
Pass context to TestOnBorrow

### ISSUE THREAD

**Issue by dcormier** [gomodule/redigo#439]
I like to add tracing to my code. This depends on being able to pass a context around. To that end, I wish `TestOnBorrow` was passed the context that's passed to `GetContext`. Perhaps this would require a new function so it's not a breaking change: `TestOnBorrowContext`, or something similar.

**Comment by stevenh** [comment]
Indeed this would require a new method.

**Comment by dcormier** [comment]
Uh, so, was a new method added? The PR was closed without merging, resulting in this issue being closed.

**Comment by stevenh** [comment]
Yes you can see the comments on the https://github.com/gomodule/redigo/pull/442

**Comment by stevenh** [comment]
@dcormier just flagging this was reopened in case you didn't see the notification.

Also mentioning that I'm thinking of a v2 which would help clean up the existing interfaces, so would be interested in peoples feedback on use cases like this we should consider when designing that.
